SourcePointMetaApp/SourcePointMetaAppTests/ConsentViewControllerTests/ConsentViewControllerErrorTest.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  ConsentViewControllerErrorTest.swift
+//  ConsentViewControllerTests
+//
+//  Created by Vilas on 9/9/19.
+//
+
+import XCTest
+@testable import ConsentViewController
+
+class ConsentViewControllerErrorTest: XCTestCase {
+    
+    var consentViewControllerError: ConsentViewControllerError?
+    
+    override func setUp() {
+        consentViewControllerError = ConsentViewControllerError()
+    }
+    
+    override func tearDown() {
+        
+    }
+    
+    /// this test method is used to test parsing error method
+    func testUnableToParseConsentStringError() {
+        let errorObject = UnableToParseConsentStringError(euConsent: "BOl9F8EOl9F8EAGABBENCi-AAAAp6ABAFIA")
+        if (errorObject.failureReason != nil) && !errorObject.description.isEmpty {
+            XCTAssert(true, "Error handling for the consent parsing failure is successfully handled")
+        } else {
+            XCTAssert(false, "Error handling for the consent parsing failure is not handled properly")
```

Comment:
We should test the actual content of the error messages. E.g. 
```
UnableToParseConsentStringError(euConsent: "BOl9F8EOl9F8EAGABBENCi-AAAAp6ABAFIA").description == "Could not parse the raw string BOl9F8EOl9F8EAGABBENCi-AAAAp6ABAFIA into a ConsentString"
```

the same is true for all other tests.

Reply:
Done. If we change the format of error description in future then we have to change test cases also thats why I was just checking whether the description is empty or not.